pkg/network/tracer/testutil/proxy/unix_transparent_proxy.go

```diff
@@ -0,0 +1,168 @@
+	useTLS bool
+	// isReady is a flag indicating whether the server is ready to accept connections.
+	isReady atomic.Bool
+	// wg is a wait group used to wait for the server to stop.
+	wg sync.WaitGroup
+	// ln is the listener used by the server.
+	ln net.Listener
+}
+
+// NewUnixTransparentProxyServer returns a new instance of a UnixTransparentProxyServer.
+func NewUnixTransparentProxyServer(unixPath, remoteAddr string, useTLS bool) *UnixTransparentProxyServer {
+	return &UnixTransparentProxyServer{
+		unixPath:   unixPath,
+		remoteAddr: remoteAddr,
+		useTLS:     useTLS,
+	}
+}
+
+// Run starts the proxy server.
+func (p *UnixTransparentProxyServer) Run() error {
+	// Clear the old socket if it exists.
+	if err := p.clearOldSocket(); err != nil {
+		return err
+	}
+
+	ln, err := net.Listen("unix", p.unixPath)
+	if err != nil {
+		return err
+	}
+	p.ln = ln
+	p.wg.Add(1)
+	go func() {
+		defer p.wg.Done()
+		p.isReady.Store(true)
+
+		for {
+			unixSocketConn, err := ln.Accept()
+			if err != nil {
+				// We can get this error when the listener is closed, during shutdown.
+				if !errors.Is(err, net.ErrClosed) {
+					log.Errorf("failed accepting connection: %s", err)
+				}
+				return
+			}
+			go p.handleConnection(unixSocketConn)
+		}
+	}()
+
+	return nil
+}
+
+// Stop stops the proxy server.
+func (p *UnixTransparentProxyServer) Stop() {
+	defer p.clearOldSocket()
+
+	_ = p.ln.Close()
+	p.wg.Wait()
+}
+
+// WaitUntilServerIsReady blocks until the server is ready to accept connections.
+func (p *UnixTransparentProxyServer) WaitUntilServerIsReady() {
+	for !p.isReady.Load() {
+		time.Sleep(10 * time.Millisecond)
+	}
+}
+
+// WaitForConnectionReady blocks until the server is ready to accept connections.
+// meant for external servers.
+func WaitForConnectionReady(unixSocket string) error {
+	for i := 0; i < 30; i++ {
+		_, err := net.DialTimeout("unix", unixSocket, time.Second)
+		if err == nil {
+			return nil
+		}
+		time.Sleep(time.Millisecond * 100)
+	}
+
+	return fmt.Errorf("could not connect %q after 30 retries", unixSocket)
+}
+
+// handleConnection handles a new connection, by forwarding all traffic to the remote address.
+func (p *UnixTransparentProxyServer) handleConnection(unixSocketConn net.Conn) {
+	defer unixSocketConn.Close()
+
+	var remoteConn net.Conn
+	var err error
+	if p.useTLS {
+		timedContext, cancel := context.WithTimeout(context.Background(), defaultDialTimeout)
+		dialer := &tls.Dialer{Config: &tls.Config{InsecureSkipVerify: true}}
+		remoteConn, err = dialer.DialContext(timedContext, "tcp", p.remoteAddr)
+		cancel()
+	} else {
+		remoteConn, err = net.DialTimeout("tcp", p.remoteAddr, defaultDialTimeout)
+	}
+	if err != nil {
+		log.Errorf("failed to dial remote: %s", err)
+		return
+	}
+	defer remoteConn.Close()
+
+	var streamWait sync.WaitGroup
+	streamWait.Add(2)
+
+	streamConn := func(dst io.Writer, src io.Reader) {
+		defer streamWait.Done()
+		io.Copy(dst, src)
```

Comment:
check the error

Reply:
ignoring it, as we have nothing to do with it